udsoncan/services/Authentication.py

```diff
@@ -0,0 +1,383 @@
+import struct
+from typing import Optional, cast
+
+from udsoncan.Request import Request
+from udsoncan.Response import Response
+from udsoncan.exceptions import *
+from udsoncan.BaseService import BaseService, BaseResponseData, BaseSubfunction
+from udsoncan.ResponseCode import ResponseCode
+import udsoncan.tools as tools
+
+
+class AuthenticationReturnParameter:
+    RequestAccepted = 0x00
+    GeneralReject = 0x01
+    AuthenticationConfiguration_APCE = 0x02
+    AuthenticationConfiguration_ACR_with_asymmetric_cryptography = 0x03
+    AuthenticationConfiguration_ACR_with_symmetric_cryptography = 0x04
+
+    DeAuthentication_successful = 0x10
+    CertificateVerified_OwnershipVerificationNecessary = 0x11
+    OwnershipVerified_AuthenticationComplete = 0x12
+    CertificateVerified = 0x13
+
+
+class Authentication(BaseService):
+    _sid = 0x29
+
+    supported_negative_response = [ResponseCode.SubFunctionNotSupported,
```

Comment:
Any of those are always supported, so they don't need to appear here.  Altough won'T cause an issue if they are

GeneralReject
ServiceNotSupported
ResponseTooLong
BusyRepeatRequest
NoResponseFromSubnetComponent
FailurePreventsExecutionOfRequestedAction
SecurityAccessDenied
RequestCorrectlyReceived_ResponsePending
ServiceNotSupportedInActiveSession
ResourceTemporarilyNotAvailable

Reply:
But no one in your list is there, did I miss one?

Reply:
oh boy. Ignore me. I am confused.